### PR TITLE
Always use correct image meta dimensions

### PR DESCRIPTION
**Pull request by jfsoul**
## What does this change?
The original motivation for this PR is to change the metadata in both `structuredData/Image.scala` and `imageFigure.scala.html`.  The image dimensions are now for the same image as the url - whereas before they could have been different (in practice they never would be due to another bug, which will be fixed in another PR).

In doing this I wanted to simplify `Profile.scala` a bit so that the logic for whether we serve _largest_ or _best_ image for a profile is in one place (and callers of the class cannot pick and choose whether to follow that logic).  That required a bit of renaming for clarity which is what the majority of changes here are.

## What is the value of this and can you measure success?
We always provide correct image metadata dimensions and serve masters if using imgix and `something else` if not.  The `something else` will be solidified in another PR, but it is intended to be the best crop from the grid for the provided profile size.

## Does this affect other platforms - Amp, Apps, etc?
Yes, AMP

## Tested in CODE?
Will do

### CODE REVIEW

File: article/app/model/structuredData/Image.scala [hunk 1]
```diff
@@ -1,16 +1,19 @@
...
-    "url" -> ImgSrc.findNearestSrc(picture.images, Item700),
-    "height" -> ImgSrc.getFallbackAsset(picture.images).fold(0)(_.height),
-    "width" -> ImgSrc.getFallbackAsset(picture.images).fold(0)(_.width)
-  )
+  def apply(picture: ImageElement): JsValue = {
+    val asset: Option[ImageAsset] = Item700.bestFor(picture.images)
+    Json.obj(
+      "@type" -> "ImageObject",
+      "url" -> JsString(Item700.bestSrcFor(picture.images).getOrElse("")),
+      "height" -> asset.fold(0)(_.height),
```
Comment (jfsoul) [inline article/app/model/structuredData/Image.scala]: Technically this isn't the true width even - it will just be `Item700.width` when imgix in use.

**Comment by paperboyo** [comment]
> We always provide correct image metadata dimensions and serve masters if using imgix and something else if not.

Just a question, as nothing is ever clear for me from looking at the code... Does "always" means "whenever we do currently" (there are [places when we should but don't](https://trello.com/c/cWSiXgfV/87-all-imagery-to-properly-use-master-assets-and-resizer-responsively) now)? Or does it truly mean "always" now?

Almost totally unrelatedly, e.g. Lightbox and other places could gain from height-bound image sizes (instead of width-bound only). Maybe as [slightly cheekily suggested](https://github.com/guardian/frontend/issues/12216#issuecomment-196869603) by @rich-nguyen? Maybe by using [the trick Oliver has found](https://github.com/guardian/frontend/pull/12811)?

File: common/app/views/support/Profile.scala [hunk 2]
```diff
@@ -231,38 +222,25 @@ object ImgSrc extends Logging with implicits.Strings {
       .mkString(", ")
   }
 
-  def srcsetForProfile(profile: Profile, imageContainer: ImageMedia, hidpi: Boolean): String = {
-    if(ImageServerSwitch.isSwitchedOn) {
```
Comment (NataliaLKB) [inline common/app/views/support/Profile.scala]: Why don't you have to check for this anymore?

**Review by NataliaLKB — APPROVED** [review]
Code looks good to me!

File: common/app/views/support/Profile.scala (hunk 2 above)
Comment (jfsoul) [inline common/app/views/support/Profile.scala]: It has been moved into `bestFor`

**Comment by jfsoul** [comment]
@paperboyo 
> Or does it truly mean "always" now?

Sadly not, only in the case of image metadata urls.  And due to another issue, they were already masters anyway - but by accident, not design.

**Pull request merged**